fedimint-api-client/src/api/iroh.rs

```diff
@@ -92,20 +111,77 @@ impl IClientConnector for IrohConnector {
             .get(&peer_id)
             .ok_or(PeerError::InvalidPeerId { peer_id })?;
 
-        let connection = match self.connection_overrides.get(&node_id) {
-            Some(node_addr) => {
-                trace!(target: LOG_NET_IROH, %node_id, "Using a connectivity override for connection");
-                self.endpoint
-                    .connect(node_addr.clone(), FEDIMINT_API_ALPN)
-                    .await
+        let mut join_set = JoinSet::new();
+        let connection_override = self.connection_overrides.get(&node_id).cloned();
+        let endpoint_stable = self.endpoint_stable.clone();
+        let endpoint_next = self.endpoint_next.clone();
+
+        join_set.spawn({
+            let connection_override = connection_override.clone();
+            async move {
+                Ok(match connection_override {
+                    Some(node_addr) => {
+                        trace!(target: LOG_NET_IROH, %node_id, "Using a connectivity override for connection");
+                        endpoint_stable
+                            .connect(node_addr.clone(), FEDIMINT_API_ALPN)
+                            .await
+                    }
+                    None => endpoint_stable.connect(node_id, FEDIMINT_API_ALPN).await,
+                }.map_err(PeerError::Connection)?.into_dyn())
+        }});
+
+        join_set.spawn(async move {
+            Ok(match connection_override {
+                Some(node_addr) => {
+                    trace!(target: LOG_NET_IROH, %node_id, "Using a connectivity override for connection");
+                    endpoint_next
+                        .connect(node_addr_stable_to_next(&node_addr), FEDIMINT_API_ALPN)
+                        .await
+                }
+                None => endpoint_next.connect(
+                        iroh_next::NodeId::from_bytes(node_id.as_bytes()).expect("Can't fail"),
+                        FEDIMINT_API_ALPN
+                    ).await,
+                }.map_err(PeerError::Connection)?.into_dyn() )
+        });
+
+        // Remember last error, so we have something to return if
```

Comment:
Hmmm... We should probably always return the "current", as that's the one that is expecting to work?

Reply:
I don't think there's a clear hierarchy, what is expected to work will change over time in a migration scenario.